CSLisp/src/core/Primitives.cs

```diff
@@ -200,12 +197,47 @@ public class Primitives
                 var member = TypeUtils.GetMemberFieldOrProp(type, memberName);
                 TypeUtils.SetValue(member, instance, targetValue.AsBoxedValue);
                 return targetValue;
-
             }), sideFx: SideFx.Possible),
-        };
 
+            new Primitive("make-vector", 1, new Function((Context ctx, Val a) => {
+                if(a.IsNumber) {
+                    return new Val( new List<Val>(Enumerable.Repeat((Val)0, a.AsInt)));
+                }
+                else if(a.IsCons) {
+                    return new Val( new List<Val>(a.AsCons.ToNativeList()));
+                }
+
+                return new Val((object)null);
```

Comment:
Please return Val.NIL instead; new Val(null) will return a boxed CLR null which does not pass the (null? x) predicate